netlib/http/http1/read.py

```diff
@@ -158,6 +170,41 @@ def connection_close(http_version, headers):
     return http_version != "HTTP/1.1" and http_version != b"HTTP/1.1"  # FIXME: Remove one case.
 
 
+def expect_http_body(request, response=None):
+    """
+        Returns:
+           A boolean indicating whether a body is expected
+    """
+
+    # Determine to expect a boddy according to
+    # http://tools.ietf.org/html/rfc7230#section-3.3
+    if not response:
+        headers = request.headers
+        response_code = None
+        is_request = True
+    else:
+        headers = response.headers
+        response_code = response.status_code
+        is_request = False
+
+    if is_request:
+        if headers.get("expect", "").lower() == "100-continue":
+            return False
+        if not "content-length" in headers:
+            return False
+    else:
+        if request.method.upper() == "HEAD":
+            return False
+        if 100 <= response_code <= 199:
+            return False
+        if response_code == 200 and request.method.upper() == "CONNECT":
+            return False
+        if response_code in (204, 304):
+            return False
+
+    return True
```

Comment:
I think this method also should work for HTTP/2. 
If so, please move it to a different file to represent this.